### PR TITLE
Push auth request

### PR DESCRIPTION
**Pull request by bradumbaugh**
Added information about requesting permission to receive push notifications.

### CODE REVIEW

**Comment by bradumbaugh** [comment]
@tapashmajumder Not sure about this build failure?

File: README.md [hunk 1]
```diff
@@ -186,9 +186,47 @@ IterableAPI.email = @"user@example.com";
...
+    [center requestAuthorizationWithOptions: (UNAuthorizationOptionAlert + UNAuthorizationOptionBadge + UNAuthorizationOptionSound)
+       completionHandler:^(BOOL granted, NSError * _Nullable error) {
+       // ...
+    }];
+    ```
+    
+    By default, if you have set `IterableAPI.email` or `IterableAPI.userId`
+    (as described above), the SDK will then register the app for remote 
+    notifications automatically.
+
```
Comment (tapashmajumder) [inline README.md]: This is a little tricky. If you set userId or email. IterableSDK will register for token without asking for permission. If the app has background mode enabled then we can send silent pushes. I am not sure if we should make this clearer.

Comment (tapashmajumder) [reply README.md]: This is a constant source of confusion.

Comment (bradumbaugh) [reply README.md]: @tapashmajumder I updated this—could you take another look?

**Comment by bradumbaugh** [comment]
@tapashmajumder 

- I added some content about `autoPushRegistration`. Let me know if this looks correct to you.
- Doing so pushed me to move more content from the "using the SDK" section into the "configuring the SDK" section. I'm trying to make the "configuring the SDK" section all about setup, and the "using the SDK" all about displaying messages, handling user interactions, etc.
- Other minor fixes throughout.
- Again, this is a fairly significant update. Apologies. Happy to talk in person about it if that's easier.

**Comment by tapashmajumder** [comment]
+1

**Pull request merged**